crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -0,0 +1,1227 @@
+                    CfiModule {
+                        cfi_status: ObjectFileStatus::from(err.as_ref()),
+                        ..Default::default()
+                    }
+                }
+            };
+            (code_id, cfi_module)
+        });
+        self.inner.extend(iter)
+    }
+
+    /// Returns a mapping of module IDs to paths that can then be loaded inside a procspawn closure.
+    fn for_processing(&self) -> Vec<(DebugId, PathBuf)> {
+        self.inner
+            .iter()
+            .filter_map(|(id, module)| Some((*id, module.cfi_path.clone()?)))
+            .collect()
+    }
+
+    /// Returns the inner Map.
+    fn into_inner(self) -> BTreeMap<DebugId, CfiModule> {
+        self.inner
+    }
+}
+
+fn object_info_from_minidump_module(ty: ObjectType, module: &MinidumpModule) -> RawObjectInfo {
+    // Some modules are not objects but rather fonts or JIT areas or other mmapped files
+    // which we don't care about.  These may not have complete information so map these to
+    // our schema by converting to None when needed.
+    let code_id = module
+        .code_identifier()
+        .filter(|code_id| !code_id.is_nil())
+        .map(|code_id| code_id.to_string().to_lowercase());
+    let code_file = module.code_file();
+    let code_file = match code_file.is_empty() {
+        true => None,
+        false => Some(code_file.into_owned()),
+    };
+
+    RawObjectInfo {
+        ty,
+        code_id,
+        code_file,
+        debug_id: module.debug_identifier().map(|c| c.breakpad().to_string()),
+        debug_file: module.debug_file().map(|c| c.into_owned()),
+        image_addr: HexValue(module.base_address()),
+        image_size: match module.size() {
+            0 => None,
+            size => Some(size),
+        },
+    }
+}
+
+async fn stackwalk_with_rust_minidump(
+    cfi_caches: Vec<(DebugId, PathBuf)>,
+    minidump_path: PathBuf,
+    spawn_time: SystemTime,
+    return_modules: bool,
+) -> Result<StackWalkMinidumpResult, ProcError> {
+    if let Ok(duration) = spawn_time.elapsed() {
+        metric!(timer("minidump.stackwalk.spawn.duration") = duration);
+    }
+
+    // Stackwalk the minidump.
+    let duration = Instant::now();
+    let minidump = Minidump::read(ByteView::open(minidump_path)?)?;
+    let provider = SymbolicatorSymbolProvider::new(cfi_caches.iter());
+    let process_state = minidump_processor::process_minidump(&minidump, &provider).await?;
+    let duration = duration.elapsed();
+
+    let minidump_state = MinidumpState::from_process_state(&process_state);
+    let object_type = minidump_state.object_type();
+
+    let missing_modules = provider.missing_ids();
+    let modules = return_modules.then(|| {
+        process_state
+            .modules
+            .iter()
+            .map(|module| {
+                (
+                    // TODO(ja): Check how this can be empty and how we shim.
+                    //           Probably needs explicit conversion from raw
```

Comment:
It can be empty because minidump modules could be things that aren't object-like, e.g. font mappings etc etc.  We probably should clean up this TODO.

Reply:
Do you mean just removing the comment or does something need to functionally change?